src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -46,89 +48,167 @@ class MatchSamplingRate(Transform):
-
+    | feature_a                   | feature_b                   | const |
+    | ---                         | ---                         | ---   |
+    | list[struct[time,struct[]]] | list[struct[time,struct[]]] | int   |
+    | --------------------------- | --------------------------- | ----- |
+    | [{12:26:00.0, {1.2}},       | [{12:26:00.0, {1.2}},       | 1     |
+    |  {12:26:01.0, {2.4}},       |  {12:26:01.0, {1.4}},       |       |
+    |  {12:26:02.0, {3.6}},       |  {12:26:02.0, {1.6}},       |       |
+    |  {12:26:03.0, {4.8}}]       |  {12:26:03.0, {1.8}}]       |       |
 
-
-    Note that the used feature `time_feature_b` is still
-    present in the DataFrame. To remove it use the `select` transform.
-    """  # noqa: E501
+    """
 
     def __init__(
         self,
-        reference_timestamps: str,
-        feature_columns_with_timestamps: dict[str, str],
+        reference_feature_name: str,
+        feature_interpolation_map: dict[str, MatchSamplingRateMethod],
     ) -> None:
-        """Initialize the MatchSamplingRate transform.
+        """Initialize the transform.
 
         Args:
-            reference_timestamps: Timestamps of the reference feature.
-            feature_columns_with_timestamps: Names of the features that are
-                getting interpolated with their respective original timestamp
-                feature names.
+            reference_feature_name: Reference timeseries feature.
+            feature_interpolation_map: Key-value pairs of the timeseries
+                features that are targeted in interpolation columns and the
+                interpolation method to use. At the moment, the interpolation
+                method can only be 'linear'.
         """
-        self.reference_timestamps = reference_timestamps
-        self.feature_columns_with_timestamps = feature_columns_with_timestamps
+        self.reference_feature_name = reference_feature_name
+        self.feature_interpolation_map = feature_interpolation_map
 
     def apply(self, data: pl.LazyFrame) -> pl.LazyFrame:
-        logger.debug("Matching sampling rate of time series.")
-
-        df = data.collect(streaming=True)
-
-        for i in range(len(df)):
-            reference_timestamps = df[self.reference_timestamps][i]
-            reference_timestamps_np = reference_timestamps.to_numpy()
-            if np.issubdtype(reference_timestamps_np.dtype, np.datetime64):
-                reference_timestamps = (
-                    reference_timestamps_np.astype("datetime64[us]").astype(
-                        np.int64,
-                    )
-                    / 1e6
+        """Transform the input DataFrame.
+
+        Args:
+            data: Input DataFrame.
+
+        Returns:
+            Transformed DataFrame.
+
+        """
+        # preserve all constant columns that are not timeseries data
+        transformed_data = pl.DataFrame()
+        for i in range(len(data.collect().rows())):
```

Comment:
Iterating rows might be extremely slow... If it is fine for now, we can keep it like that. Alternatively we should try to use the expression syntax, if that is possible